### PR TITLE
Add distance class and distance_between_air method

### LINKED ISSUE #13: Add coordinates data to cities and distance calculations

### PR DESCRIPTION
**Pull request by sarslanoglu**
### Related github issue for this PR

Resolves #13 

### Checklist

* [x] I have performed a self-review of my own code
* [x] I added comments, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation `README.md`
* [x] I added tests that prove my fix is effective or that my feature works
* [x] `bundle exec rake` passes locally
* [x] `rubocop` gives no error locally
* [x] My PR title includes "WIP" or is [draft PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) if work is in progress

### Description

First part of the long waited feature distance calculations are added with this PR. Air travel distance and time estimations are added to the codebase.

### Type of change

* New feature (non-breaking change which adds functionality)

### Which tests are written for this PR? And what are the expected results for these tests?

Tests under `describe '#distance_between' do`

### CODE REVIEW

File: spec/turkish_cities_spec.rb [hunk 1]
```diff
@@ -219,6 +219,48 @@
...
+        expect(very_short_distance_results[2])
+          .to eq 'Air travel distance between Bolu and Kastamonu is 195.4 km. '\
+            'Estimated air travel would take 49 minutes.'
+        short_distance_results = TurkishCities.distance_between('izmir', 'Antalya', 'air')
+        expect(short_distance_results[0]).to eq 357.18
+        expect(short_distance_results[1]).to eq 65
+        expect(short_distance_results[2])
+          .to eq 'Air travel distance between İzmir and Antalya is 357.18 km. '\
+            'Estimated air travel would take 65 minutes.'
+        medium_distance_results = TurkishCities.distance_between('istanbul', 'Ordu', 'air')
```
Comment (sarslanoglu) [inline spec/turkish_cities_spec.rb:237]: 😉 @enderahmetyurt

File: spec/turkish_cities_spec.rb [hunk 2]
```diff
@@ -219,6 +219,48 @@
     end
   end
 
+  describe '#distance_between' do
+    context 'when input is supported' do
+      it 'finds distance between two location' do
+        very_short_distance_results = TurkishCities.distance_between('Bolu', 'Kastamonu', 'air')
```
Comment (sarslanoglu) [inline spec/turkish_cities_spec.rb:225]: 🙃 @serpilacar

File: lib/turkish_cities/distance.rb [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    # TODO
+  end
+
+  def distance_between_air(from, to)
+    city_array = find_city_attributes(from, to)
+    case city_array
+    when String
+      city_array
+    when Array
+      result = []
```
Comment (enderahmetyurt) [inline lib/turkish_cities/distance.rb:41]: `results` daha okunur olurmus sanki.

File: lib/turkish_cities/distance.rb [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  CITY_LIST = YAML.load_file(file_path)
+
+  def distance_between(from, to, travel_method)
+    case travel_method
+    when 'land'
+      distance_between_land(from, to)
+    when 'sea'
+      distance_between_sea(from, to)
+    else
+      distance_between_air(from, to)
```
Comment (enderahmetyurt) [inline lib/turkish_cities/distance.rb:21]: bence burada `else` de `air` demek cok guzel degil gibi. cunku methodun adi `distance_between` yani bir `travel_method` aliyor ve ona gore bi method cagriyor ama `else` de air var. sanki orada error raise etsek `when air` ile de `distance_between_air` cagrilsa daha tatli gibi. default'da `air` olur simdilik. ne dersin?

File: lib/turkish_cities/distance.rb [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+require_relative '../turkish_cities/helpers/decomposer_helper'
+
+class Distance
+  include DecomposerHelper
+
+  file_path = File.join(File.dirname(__FILE__), 'data/cities.yaml')
+  CITY_LIST = YAML.load_file(file_path)
+
+  def distance_between(from, to, travel_method)
```
Comment (enderahmetyurt) [inline lib/turkish_cities/distance.rb:14]: bu `from` ve `to`'yu her method'da vermek yerine class'i initlesek ve orada attribute olarak olsa mesela.

```ruby
distance = Distance.new('Istanbul', 'Ordu')
distance.distance_between_air
```

gibi olsa mi?

File: lib/turkish_cities/distance.rb [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    lat_diff = degree_to_radian(city_array[1]['latitude'] - city_array[0]['latitude'])
+    lng_diff = degree_to_radian(city_array[1]['longitude'] - city_array[0]['longitude'])
+
+    from_lat = degree_to_radian(city_array[0]['latitude'])
+    to_lat = degree_to_radian(city_array[1]['latitude'])
+
+    cosines_var = (Math.sin(lat_diff / 2)**2) + ((Math.sin(lng_diff / 2)**2) * (Math.cos(from_lat) * Math.cos(to_lat)))
+    result = 2 * Math.atan2(Math.sqrt(cosines_var), Math.sqrt(1 - cosines_var))
+    # Earth radius in km is 6371
+    6371 * result
```
Comment (enderahmetyurt) [inline lib/turkish_cities/distance.rb:75]: istersen bunu constant yapalim mi?

File: lib/turkish_cities/distance.rb [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  end
+  # rubocop:enable Metrics/AbcSize
+
+  def distance_time_estimation(air_distance)
```
Comment (enderahmetyurt) [inline lib/turkish_cities/distance.rb:79]: bu method ne yapiyor belki bi comment koymak gerek. pek anlamadim oradaki islemleri.

File: spec/turkish_cities_spec.rb [hunk 8]
```diff
@@ -219,6 +219,48 @@
     end
   end
 
+  describe '#distance_between' do
```
Comment (enderahmetyurt) [inline spec/turkish_cities_spec.rb:222]: `air` disinda sacma bisi gelirse ne oluyor? belki onun da testini eklemek lazim.

**Review by enderahmetyurt — APPROVED** [review]

**Review by serpilacar — APPROVED** [review]

**Pull request merged**